vnext/Microsoft.ReactNative.Cxx/TurboModuleProvider.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <JSI/JsiAbiApi.h>
+#include <JSI/JsiApiContext.h>
+#include <ReactCommon/TurboModule.h>
+#include <winrt/Microsoft.ReactNative.h>
+#include <winrt/Windows.Foundation.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+// Creates CallInvoker based on JSDispatcher.
+std::shared_ptr<facebook::react::CallInvoker> MakeAbiCallInvoker(IReactDispatcher const &jsDispatcher) noexcept;
+
+template <
+    typename TTurboModule,
+    std::enable_if_t<std::is_base_of_v<::facebook::react::TurboModule, TTurboModule>, int> = 0>
+void AddTurboModuleProvider(IReactPackageBuilder const &packageBuilder, std::wstring_view moduleName) {
+  auto experimental = packageBuilder.as<IReactPackageBuilderExperimental>();
+  experimental.AddTurboModule(
+      moduleName, [](IReactModuleBuilder const &moduleBuilder) noexcept->winrt::Windows::Foundation::IInspectable {
+        IJsiHostObject abiTurboModule{nullptr};
+        // We expect the initializer to be called immediately for TurboModules
```

Comment:
>immediately for TurboModules [](start = 50, length = 28)

(nit) Should we add some sort of check for that?

Reply:
No, this is an internal guarantee. If it is not met, then we will have a crash here. I would prefer not to complicate code here.